### PR TITLE
feat: add star-list command to import taps from GitHub star lists (#25)

### PR DESCRIPTION
**Pull request by EYH0602**
## Summary

Add the `star-list` command that imports all repositories from a GitHub star list as taps, enabling bulk skill discovery and installation from curated lists.

### What's new

- **`skillshub star-list <url>`** — parses a GitHub star list URL (`https://github.com/stars/<user>/lists/<list-name>`), fetches all repositories via the GraphQL API, and adds each one as a tap
- **`--install` flag** — optionally installs all skills from each newly added tap in one step
- **URL parsing** (`parse_star_list_url`) — validates and extracts username + list name from various URL formats (https, http, no protocol, trailing slash)
- **GraphQL fetching** (`fetch_star_list_repos`) — queries the GitHub GraphQL API with:
  - Two-step lookup: first finds the list ID by name, then fetches repos from that list
  - Cursor-based pagination for lists with many repos
  - Retry logic with exponential backoff for transient errors and rate limits
  - Requires `GITHUB_TOKEN` (GraphQL API is authenticated-only)
- **Import orchestration** (`import_star_list` in `tap.rs`) — iterates fetched repos, skips already-added taps, reports added/skipped/failed counts

### Files changed

| File | Change |
|------|--------|
| `src/cli.rs` | Add `StarList` command variant with `url` and `--install` args |
| `src/main.rs` | Wire `StarList` command to `tap::import_star_list` |
| `src/registry/github.rs` | Add `parse_star_list_url`, `fetch_star_list_repos`, and 10 unit tests |
| `src/registry/tap.rs` | Add `import_star_list` function |
| `src/registry/mod.rs` | Re-export new public items |
| `Cargo.toml` / `Cargo.lock` | Version bump |
| `README.md` / `AGENTS.md` | Document the new command |

### Usage

```bash
# Add all repos from a star list as taps
skillshub star-list https://github.com/stars/EYH0602/lists/skills

# Add taps and install all their skills
skillshub star-list https://github.com/stars/EYH0602/lists/skills --install
```

### Test plan

- [x] Unit tests for `parse_star_list_url` (valid URLs, invalid URLs, edge cases)
- [x] Unit tests for `fetch_star_list_repos` with mock GraphQL server (success + list-not-found)
- [x] Manual: `skillshub star-list <url>` adds taps correctly
- [x] Manual: `skillshub star-list <url> --install` adds taps and installs skills
- [x] Manual: re-running skips already-added taps

### CODE REVIEW

File: src/registry/github.rs [hunk 1]
```diff
@@ -660,9 +665,205 @@ fn copy_dir_contents(src: &Path, dst: &Path) -> Result<()> {
...
+        }"#,
+        "variables": {
+            "login": username
+        }
+    });
+
+    let resp = send_with_retry(
+        || {
+            let req = client.post(&gql_url).json(&find_list_query);
+            require_auth(req).unwrap_or_else(|_| client.post(&gql_url))
```
Comment (EYH0602) [inline src/registry/github.rs]: **Bug: `require_auth` fallback silently drops error + request body**

If `GITHUB_TOKEN` is unset, `require_auth` returns an error. The `unwrap_or_else` fallback silently swallows that error and creates a **new** `client.post(...)` **without** `.json(&find_list_query)` — so the request body is lost. The user gets a confusing HTTP error from GitHub instead of the clear "GITHUB_TOKEN is required" message that `require_auth` was designed to provide.

Same issue on line 824 for the repos query.

**Suggested fix**: Validate the token once before entering the retry loops, then use it directly:

```rust
pub fn fetch_star_list_repos(username: &str, list_name: &str) -> Result<Vec<String>> {
    let token = std::env::var("GITHUB_TOKEN").with_context(|| {
        "GITHUB_TOKEN is required for star list operations.\n\
         The GraphQL API does not support unauthenticated requests.\n\
         Set GITHUB_TOKEN with a personal access token."
    })?;
    let client = build_client()?;
    let gql_url = graphql_url();

    // ... then in the retry closures:
    let resp = send_with_retry(
        || client.post(&gql_url).json(&find_list_query).bearer_auth(&token),
        &gql_url,
    )?;
```

This eliminates the `require_auth` helper entirely (it's only used here) and avoids the impossible fallback path.
